File: Data Structures & Algorithms/reverse-bits/submission-0.go
func reverseBits(n int) int {
	result := 0
	i := 32.0
	for n > 0 {
		i = i - 1
		result += int(math.Pow(2,i)) * (n%2)
		n=n>>1
	}
	return result
}

File: Data Structures & Algorithms/house-robber/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    
    robHouse(nums, houseNumber, moneyMap) {
        if (houseNumber >= nums.length) {
            return 0
        }
        if (moneyMap.has(houseNumber)) {
            return moneyMap.get(houseNumber)
        }
        const result = Math.max(
            nums[houseNumber] + this.robHouse(nums, houseNumber + 2, moneyMap),
            this.robHouse(nums, houseNumber + 1, moneyMap)
            )
        moneyMap.set(houseNumber, result)
        return result
    }

    rob(nums) {
        const maxMoney = new Map();
        return this.robHouse(nums, 0, maxMoney)
    }
}
